lib/sessions.ts

```diff
@@ -85,6 +109,98 @@ export class Sessions {
     });
   }
 
+  /** Parse a JWT and verify the signature, preferring local verification over remote.
+   *
+   * If maxTokenAge is set, remote verification will be forced if the JWT was issued at (based on
+   * the "iat" claim) more than maxTokenAge seconds ago.
+   */
+  async authenticate_jwt(
+    jwt: string,
+    options?: {
+      maxTokenAge?: number; // seconds
+    }
+  ): Promise<AuthenticateResponse> {
+    try {
+      const session = await this.authenticate_jwt_local(jwt, options);
+      return {
+        session,
+        session_jwt: jwt,
+        session_token: "",
+
+        // Pretend that this was a successful API request.
+        request_id: "",
+        status_code: 200,
+      };
+    } catch (err) {
+      if (err instanceof ClientError && err.code === "jwt_too_old") {
+        // Token was too old (stale) to verify locally. Check with the Stytch API.
+        return this.authenticate({ session_jwt: jwt });
+      }
+
+      throw err;
+    }
+  }
+
+  /** Parse a JWT and verify the signature locally (without making an /authenticate call).
+   *
+   * If maxTokenAge is set, this will return an error if the JWT was issued (based on the "iat"
+   * claim) more than maxTokenAge seconds ago.
+   */
+  async authenticate_jwt_local(
+    jwt: string,
+    options?: {
+      maxTokenAge?: number;
+    }
+  ): Promise<Session> {
+    // Don't pass maxTokenAge directly to jwtVerify because it interprets zero as "infinity". We
+    // want zero to mean "every token is stale" and force remote verification.
+    const maxTokenAge = options?.maxTokenAge;
+    const now = Date.now() / 1000; // Unix epoch seconds
+
+    let payload;
+    try {
+      const token = await jose.jwtVerify(jwt, this.jwks, this.jwtOptions);
+      payload = token.payload;
+    } catch (err) {
+      throw new ClientError("jwt_invalid", "Could not verify JWT", err);
+    }
+
+    if (maxTokenAge != null) {
+      const iat = payload.iat;
+      if (!iat) {
+        throw new ClientError("jwt_invalid", "JWT was missing iat claim");
+      }
+      if (now - iat > maxTokenAge) {
+        throw new ClientError(
+          "jwt_too_old",
+          `JWT was issued at ${iat}, more than ${maxTokenAge} seconds ago`
+        );
+      }
+    }
+
+    // Re-pack the JWT contents as session data. Since we're actually changing the types of
+    // values stored on the session object, we have to disable type-checking for a bit.
+    //
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    const session = payload[sessionClaim] as any;
+
+    // The subject claim is the user ID.
+    session.user_id = payload.sub;
```

Comment:
and if you are set on mutating in place, clear up the `sub` here like you clear up the `id` below